internal/controller/nimservice_controller.go

```diff
@@ -294,60 +292,33 @@ func (r *NIMServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 		)
 
-	nimServiceBuilder, err = k8sutil.ControllerCallbackIfCRDExists(r.discoveryClient,
-		nimServiceBuilder,
-		resourcev1beta2.SchemeGroupVersion.WithResource("resourceclaims"),
-		func(bd *builder.Builder) *builder.Builder {
-			return bd.Watches(
-				&resourcev1beta2.ResourceClaim{},
-				handler.EnqueueRequestsFromMapFunc(r.mapResourceClaimToNIMService),
-				builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
-			)
-		},
-	)
+	resourceClaimCRDExists, err := k8sutil.CRDExists(r.discoveryClient, resourcev1beta2.SchemeGroupVersion.WithResource("resourceclaims"))
 	if err != nil {
 		return err
 	}
-
-	nimServiceBuilder, err = k8sutil.ControllerOwnsIfCRDExists(
-		r.discoveryClient,
-		nimServiceBuilder,
-		resourcev1beta2.SchemeGroupVersion.WithResource("resourceclaimtemplates"),
-		&resourcev1beta2.ResourceClaimTemplate{},
-	)
-	if err != nil {
-		return err
+	if resourceClaimCRDExists {
```

Comment:
similarly we need to add for resourceclaimtemplates too, i prefer to keep the util method as-is and just revert the httproute checks.